hazelcast/src/hazelcast/client/compact.cpp

```diff
@@ -1694,28 +1857,120 @@ schema_writer::build() &&
     return schema{ type_name, std::move(field_definition_map) };
```

Comment:
Java has this check, can you add it? 
```
        if (!fieldNames.add(fieldName)) {
            throw new HazelcastSerializationException("Field with the name '" + fieldName + "' already exists");
        }
```

Reply:
This check and unit test is added which tests this, `CompactSerializationTest.test_write_to_same_field_twice`

Fixed.